utxo.go

```diff
@@ -0,0 +1,193 @@
+package dcrlibwallet
+
+import (
+	"fmt"
+
+	"github.com/decred/dcrd/chaincfg/v2"
+	"github.com/decred/dcrd/dcrutil/v2"
+	"github.com/decred/dcrd/txscript"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/dcrwallet/errors/v2"
+	"github.com/decred/dcrwallet/wallet/v3/txauthor"
+	"github.com/decred/dcrwallet/wallet/v3/txrules"
+	"github.com/decred/dcrwallet/wallet/v3/txsizes"
+	"github.com/planetdecred/dcrlibwallet/txhelper"
+)
+
+type nextAddressFunc func() (address string, err error)
+
+func calculateChangeScriptSize(changeAddress string, chainParams *chaincfg.Params) (int, error) {
+	changeSource, err := txhelper.MakeTxChangeSource(changeAddress, chainParams)
+	if err != nil {
+		return 0, fmt.Errorf("change address error: %v", err)
+	}
+	return changeSource.ScriptSize(), nil
+}
+
+func calculateMultipleChangeScriptSize(changeDestinations []TransactionDestination,
+	chainParams *chaincfg.Params) (int, error) {
+	var totalChangeScriptSize int
+	for _, changeDestination := range changeDestinations {
+		changeScriptSize, err := calculateChangeScriptSize(changeDestination.Address, chainParams)
+		if err != nil {
+			return 0, err
+		}
+		totalChangeScriptSize += changeScriptSize
+	}
+	return totalChangeScriptSize, nil
+}
+
+// ParseOutputsAndChangeDestination generates and returns TxOuts
+// using the provided slice of transaction destinations.
+// Any destination set to receive max amount is not included in the TxOuts returned,
+// but is instead returned as a change destination.
+// Returns an error if more than 1 max amount recipients identified or
+// if any other error is encountered while processing the addresses and amounts.
+func (tx *TxAuthor) ParseOutputsAndChangeDestination(txDestinations []TransactionDestination) ([]*wire.TxOut, int64, string, error) {
+	var outputs = make([]*wire.TxOut, 0)
+	var totalSendAmount int64
+	var maxAmountRecipientAddress string
+
+	for _, destination := range txDestinations {
+		// validate the amount to send to this destination address
+		if !destination.SendMax && (destination.AtomAmount <= 0 || destination.AtomAmount > dcrutil.MaxAmount) {
```

Comment:
Don't do amount validation here, do it when adding the send destination [here](https://github.com/planetdecred/dcrlibwallet/blob/f1488744385e850f0f411de7508edd0598e81879/txauthor.go#L39) and [here](https://github.com/planetdecred/dcrlibwallet/blob/f1488744385e850f0f411de7508edd0598e81879/txauthor.go#L54)

Reply:
On the mobile or wallet if developer is forget to check errors at `AddSendDestination` or `UpdateSendDestination`, we should validate again here, right?